### PR TITLE
Support server side encryption in CollectiveStorage

### PR DESCRIPTION
**Pull request by mejo-**
### 📝 Summary

Fixes: #285

Tested scenarios:
* [x] Create collective unencrypted, enable encryption, add pages
* [x] Copy encrypted page from encrypted collective to encrypted home
* [x] Copy encrypted page from encrypted collective to unencrypted home

### CODE REVIEW

**Comment by mejo-** [comment]
Something else to decide on: do we want server side encryption to be always enabled for Collectives when it's enabled on the instance, or shall there be an option to disable it in Collectives separately. I don't see a strong reason for this. Groupfolders even has it disabled per default, though: https://github.com/nextcloud/groupfolders/pull/2057/commits/2df87605013250c741ea578a83012d172f30f5c0

**Comment by mejo-** [comment]
> Might be good to have some set of e2e/integration tests also running against this

Thanks, good idea. Added a new `behat-sqlite-encryption` workflow for this now.

File: docs/content/administration/_index.md [hunk 1]
```diff
@@ -21,6 +21,14 @@ The **Collectives** app can be installed from the [Nextcloud App Store](https://
 In your Nextcloud instance, simply navigate to **»Apps«**, find the
 **»Collectives«** app and enable it.
 
+## Collectives and server-side encryption
+
+With server-side encryption enabled, the files in a Collective will be stored
+encrypted on the filesystem as well.
```
Comment (juliushaertl) [inline docs/content/administration/_index.md:27]: Maybe even reference https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/encryption_configuration.html#files-not-encrypted here as previews would also be relevant for collectives

**Comment by mejo-** [comment]
@juliushaertl I did some further testing. `encryption:decrypt-all` and `encryption:encrypt-all` works as expected, no need for further work here. The PR is ready to be merged from my side. Could you take another look and approve?

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**